src/rfbzero/degradation.py

```diff
@@ -273,5 +273,62 @@ def degrade(self, c_ox: float, c_red: float, timestep: float) -> tuple[float, fl
             c_ox, c_red = mechanism.degrade(c_ox, c_red, timestep)
         return c_ox, c_red
 
-# TODO
-# def potential_dependent(potential, c_ox, c_red, timestep, rate=0):
+
+class Dimerization(DegradationMechanism):
+    """
+    Provides a reversible dimerization mechanism:
+        ox + red <--> dimer
+
+    Parameters
+    ----------
+    forward_rate_constant : float
+        Second order rate constant for forward reaction (1/M/s).
+    backward_rate_constant : float
+        First order rate constant for backward reaction (1/s).
+    c_dimer : float
+        Initial concentration of dimer, defaults to 0 (M).
+        Continuously updated.
+
+    """
+
+    def __init__(self, forward_rate_constant: float, backward_rate_constant: float, c_dimer: float = 0.0):
+        self.forward_rate_constant = forward_rate_constant
+        self.backward_rate_constant = backward_rate_constant
+        self.c_dimer = c_dimer
+
+        if self.forward_rate_constant <= 0.0 or self.backward_rate_constant <= 0.0:
+            raise ValueError("'forward/backward_rate_constant' must be > 0.0")
+
```

Comment:
Do we also want to ensure that `forward_rate_constant > backward_rate_constant`?

Reply:
No, this is also a valid case. It would result in minimal---but non-zero--- amounts of dimer formed.